lib/ocpp/v201/smart_charging.cpp

```diff
@@ -192,12 +192,23 @@ SmartChargingHandler::validate_profile_schedules(ChargingProfile& profile,
     return ProfileValidationResultEnum::Valid;
 }
 
-void SmartChargingHandler::add_profile(int32_t evse_id, ChargingProfile& profile) {
+SetChargingProfileResponse SmartChargingHandler::add_profile(int32_t evse_id, ChargingProfile& profile) {
     if (STATION_WIDE_ID == evse_id) {
         station_wide_charging_profiles.push_back(profile);
     } else {
         charging_profiles[evse_id].push_back(profile);
     }
+    SetChargingProfileResponse response;
+    response.status = ChargingProfileStatusEnum::Accepted;
+    return response;
+}
+
+std::vector<ChargingProfile> SmartChargingHandler::get_profiles() {
+    auto all_profiles = station_wide_charging_profiles;
```

Comment:
So this is me being unfamiliar with C++, but: is `station_wide_charging_profiles` being copied here, or do modifications to `all_profiles` modify `station_wide_charging_profiles`?